firebase-firestore/src/main/java/com/google/firebase/firestore/util/CustomClassMapper.java

```diff
@@ -119,9 +119,13 @@ public static Object convertToPlainJavaTypes(Object object) {
         throw serializeError(path, "Shorts are not supported, please use int or long");
       } else if (o instanceof Byte) {
         throw serializeError(path, "Bytes are not supported, please use int or long");
-      } else {
-        // Long, Integer, Double
+      } else if (o instanceof Long || o instanceof Integer || o instanceof Double) {
```

Comment:
The underlying UserDataConverter handles `Float` types so you can combine the cases that return a value. With this general error message it no longer seems worthwhile to handle `Short` and `Byte` separately.

Something like this:

```java
if (o instanceof Long || o instanceof Integer || o instanceof Double || o instanceof Float) {
  return o;
} else {
  throw serializeError(...);  // your new message
}
```

Reply:
Yeah, that is a good suggestion. I moved some of the code duplication to the test cases instead...